addons/mail/static/src/model/model_core.js

```diff
@@ -341,16 +348,29 @@ export function patchRecordMethods(modelName, recordMethods) {
 }
 
 /**
- * @param {Object} definition The JSON definition of the model to register.
- * @param {Object} [definition.fields]
- * @param {string[]} definition.identifyingFields
- * @param {Object} [definition.lifecycleHooks]
- * @param {Object} [definition.modelGetters] Deprecated; use fields instead.
- * @param {Object} [definition.modelMethods]
- * @param {string} definition.name
- * @param {OnChanges[]} [definition.onChanges]
- * @param {Object} [definition.recordGetters] Deprecated; use fields instead.
- * @param {Object} [definition.recordMethods]
+ * @typedef {Object} ModelDefinition
+ * @property {Object} [fields] Fields of the model.
+ * Keys are the name of the field, value are the field definition
+ * @property {Array<string|string[]>} identifyingFields Fields used to uniquely identify a record of this model.
+ * - [field1, fields2] means field1 AND field2
+ * - [[field1], [field2]] means field1 OR field2.
+ * - [[field1], [field2, fields3]] means field1 AND (field2 OR field3).
+ * - [] means singleton
+ * @property {LifecycleHookDefinition} [lifecycleHooks] Define lifecycle hooks.
+ * @property {Object} [modelGetters] Deprecated; use fields instead.
+ * @property {Object} [modelMethods] Static methods of the model.
+ * - Keys are the name dans value are the function
+ * @property {string} name Name of the model.
+ * @property {import("@mail/model/model_onchange").OnChange[]} [onChanges] Register OnChange listenner
+ * @property {Object} [recordGetters] Deprecated; use fields instead.
+ * @property {Object} [recordMethods] Record methods of this model.
+ * - Keys are the name dans value are the function
+ */
+
+/**
+ * Register a new model.
```

Comment:
It actually registers a model definition and not a model itself.

The actual model is generated later from definitions.

Reply:
Generally speaking I agree.
But for now let's say it's a simplification to type less when using it...

No reason to not be explicit in the doc though.